qupath-gui-fx/src/main/java/qupath/lib/gui/scripting/languages/ScriptLanguage.java

```diff
@@ -0,0 +1,88 @@
+/*-
+ * #%L
+ * This file is part of QuPath.
+ * %%
+ * Copyright (C) 2014 - 2016 The Queen's University of Belfast, Northern Ireland
+ * Contact: IP Management (ipmanagement@qub.ac.uk)
+ * Copyright (C) 2018 - 2022 QuPath developers, The University of Edinburgh
+ * %%
+ * QuPath is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * 
+ * QuPath is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License 
+ * along with QuPath.  If not, see <https://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package qupath.lib.gui.scripting.languages;
+
+/**
+ * Abstract class to represent languages supported by the script editor.
+ * @author Melvin Gelbard
+ * @since v0.4.0
+ */
+public abstract class ScriptLanguage {
+	
+	protected String name;
+	protected String[] exts;
+	protected ScriptSyntax syntax;
+	protected ScriptAutoCompletor completor;
+	
+	/**
+	 * Default constructor for a {@link ScriptLanguage}.
+	 * @param name		the language name
+	 * @param exts			the possible extensions for this language
+	 * @param syntax		the syntax object for this language
+	 * @param completor	the auto-completion object for this language
+	 */
+	protected ScriptLanguage(String name, String[] exts, ScriptSyntax syntax, ScriptAutoCompletor completor) {
```

Comment:
I think `ScriptSyntax` and `ScriptAutoCompletor` shouldn't be required here - don't store them in the abstract class, and allow subclasses to return them.

Main reason: subclasses might want to initialise them within the constructor. But the `String` for name and extensions should always be available, so it's ok to make them required by the constructor.

With the current implementation, I don't think this needs to be an abstract class at all (?)